lib/rucio/transfertool/fts3.py

```diff
@@ -399,16 +436,36 @@ def build_job_params(
         elif 'default' in max_time_in_queue:
             job_params['max_time_in_queue'] = max_time_in_queue['default']
 
-    overwrite_hop = True
-    for transfer_hop in transfer_path[:-1]:
-        # Only allow overwrite if all hops in multihop allow it
-        h_overwrite = transfer_hop.rws.attributes.get('overwrite', True)
-        job_params['overwrite'] = h_overwrite and job_params['overwrite']
-        # Allow overwrite_hop if all intermediate hops allow it (ignoring the last hop)
-        overwrite_hop = h_overwrite and overwrite_hop
+    # Refer to https://its.cern.ch/jira/browse/FTS-1749 for full details (login needed), extract below:
+    # Why the overwrite_hop parameter is needed?
+    # Rucio decides that a multihop transfer is needed DISK1 --> DISK2 --> TAPE1 in order to put the file on tape. For some reason, the file already exists on DISK2.
+    # Rucio doesn't know about this file on DISK2. It could be either a correct or corrupted file. This can be due to a previous issue on Rucio side, FTS side, network side, etc (many possible reasons).
+    # Normally, Rucio allows overwrite towards any disk destination, but denies overwrite towards a tape destination. However, in this case, because the destination of the multihop is a tape, DISK2 cannot be overwritten.
+    # Proposed solution
+    # Provide an --overwrite-hop submission option, which instructs FTS to overwrite all transfers except for the destination within a multihop submission.
+
+    # direct transfers, is not multihop
+    if len(transfer_path) == 1:
+        overwrite_hop = False
+
+    else:
+        overwrite_hop = True
+        for transfer_hop in transfer_path[:-1]:
+            # Only allow overwrite if all hops in multihop allow it
+            h_overwrite = transfer_hop.rws.attributes.get('overwrite', True)
+
+            # if one hop cannot be overwritten, quit early
+            if h_overwrite is False:
+                break
+
+            job_params['overwrite'] = h_overwrite and job_params['overwrite']
+            # Allow overwrite_hop if all intermediate hops allow it (ignoring the last hop)
+            overwrite_hop = h_overwrite and overwrite_hop
+
     if not job_params['overwrite'] and overwrite_hop:
         job_params['overwrite_hop'] = overwrite_hop
```

Comment:
Out of scope of this PR, but I'm a bit confused by this condition. It only evaluates to `True` when:
- `job_params['overwrite']` is `False`
- `overwrite_hop` is `True`

I don't understand what this means in practice - do you know more about this?

Reply:
I think it means that if there is a need in a multi-hop transfer to overwrite the intermediate location and overwrite is to false (default for tape), the overwrite_hop is set to true so FTS can overwrite only the intermediate data but not the final one. I think this use case is there to allow transfers to tape that are multi-hop, and when one of the hops in the middle gets stuck, to create another request to overwrite these bad data and make it to the end hop.